plugins/couchbase_logs.yaml

```diff
@@ -0,0 +1,253 @@
+version: 0.0.1
+title: Couchbase
+description: Log parser for Couchbase
+parameters:
+  - name: enable_error_log
+    type: bool
+    default: true
+  - name: error_log_path
+    type: "[]string"
+    default:
+      - "/opt/couchbase/var/lib/couchbase/logs/error.log"
+  - name: enable_debug_log
+    type: bool
+    default: false
+  - name: debug_log_path
+    type: "[]string"
+    default: 
+      - "/opt/couchbase/var/lib/couchbase/logs/debug.log"
+  - name: enable_info_log
+    type: bool
+    default: false
+  - name: info_log_path
+    type: "[]string"
+    default: 
+      - "/opt/couchbase/var/lib/couchbase/logs/info.log"
+  - name: enable_access_log
+    type: bool
+    default: true
+  - name: http_access_log_path
+    type: "[]string"
+    default: 
+      - "/opt/couchbase/var/lib/couchbase/logs/http_access.log"
+  - name: enable_internal_access_log
+    type: bool
+    default: false
+  - name: http_internal_access_log_path
+    type: "[]string"
+    default: 
+      - "/opt/couchbase/var/lib/couchbase/logs/http_access_internal.log"
+  - name: enable_babysitter_log
+    type: bool
+    default: true
+  - name: babysitter_log_path
+    type: "[]string"
+    default: 
+      - "/opt/couchbase/var/lib/couchbase/logs/babysitter.log"
+  - name: enable_xdcr_log
+    type: bool
+    default: false
+  - name: xdcr_log_path
+    type: "[]string"
+    default: 
+      - "/opt/couchbase/var/lib/couchbase/logs/goxdcr.log"
+  - name: start_at
+    type: string
+    supported:
+      - beginning
+      - end
+    default: end
+template: |
+  receivers:
+    {{ if .enable_error_log}}
+    filelog/error_logs:
+      include: 
+        {{ range $fp := .error_log_path }}
+        - '{{ $fp }}'
+        {{ end }}
+      multiline: 
+        line_start_pattern: '^\[[^:]*:[^,]*,\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d+-\d{2}:\d{2},[^@]*@[^:]*:'
+      start_at: {{ .start_at }}
+      attributes:
+        log_type: 'couchbase.error'
+        plugin_id: 'Couchbase'
+      operators:
+        - type: regex_parser
+          regex: '^\[(?P<type>[^:]*):(?P<couchbase_severity>[^,]*),(?P<timestamp>\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d+-\d{2}:\d{2}),(?P<node>[^@]*)@(?P<host>[^:]*):(?P<source>[^\]]+)\](?P<message>(?s).*[^\n\s])[\n\s]*$'
+          timestamp:
+            parse_from: attributes.timestamp
+            layout: '%Y-%m-%dT%H:%M:%S.%L%j'
+          severity:
+            parse_from: attributes.couchbase_severity
```

Comment:
Do we know what the severities here actually are?

We implicitly map the aliases to themselves (it's the "default" preset). See https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/stanza/docs/types/severity.md#how-severity-mapping-works for the aliases, these are what automatically map.

Problem is, the aliases have changed between Stanza and OTel. So, the alias in Stanza is "warning", but in OTel it's "warn".

My concern is that this severity can be "WARNING", which will not properly map to an OTel severity, but I'm not sure whether this is the case or not.

Reply:
I don't think this is necessarily the case, I did a little bit of research and the Couchbase [Documentation](https://docs.couchbase.com/server/current/manage/manage-logging/manage-logging.html#changing-log-file-levels) indicates that default logging levels is either `debug` or `info`. 

Now there is a case I've seen where Couchbase Techonologies, such as SDK, Mobile, connectors, etc. can save logs from outside the Couchbase cluster that sometimes have a log level of Warning. [here](https://docs.couchbase.com/server/current/manage/manage-logging/manage-logging.html#redacting-log-files-outside-the-cluster)

Likewise with `couchdb.log` file in which they may have a log level of `error` but as specified in the [Couchbase Scope Doc](https://docs.google.com/document/d/1kLox-SyYwak47ORWEQvx9pDhxviixEJSArppAC2-Tf0/edit#), 
```
Note: The "<0.277.0>:couch_log:error:33]" part is omitted in the mapping for the first cut. This is what Couchbase's own Logging UI shows as well. In case they reveal a clear mapping, and there is a need for us to add it, we can add it in the future in a non-breaking way.
```

But since were not monitoring logs from `couchdb.log` I think it would be safe for now to omit the "WARNING" and allow the plugin to map the severity.

Reply:
Cool, sounds good.